src/xdg.rs

```diff
@@ -221,42 +208,125 @@ impl Icon {
+                    Bucket::Scalable
+                } else if file_name == "symbolic" {
+                    Bucket::Symbolic
+                } else if let [width, height] = file_name.split('x').collect::<Vec<_>>()[..] {
+                    match (u32::from_str(width), u32::from_str(height)) {
+                        (Ok(width), Ok(height)) if width == height => Bucket::Exact(width),
+                        _ => continue,
+                    }
+                } else {
+                    continue;
+                };
+
+                let mut dir_path = dir_entry.path();
+                dir_path.push("apps");
+                let read_dir = match fs::read_dir(dir_path) {
+                    Ok(read_dir) => read_dir,
+                    Err(_) => continue,
+                };
+                for file in read_dir {
+                    let file = file.unwrap();
+
+                    let file_name = file.file_name();
+                    let file_name = match file_name.to_str() {
+                        Some(file_name) => file_name,
+                        None => continue,
+                    };
+                    let name = file_name.rsplit_once('.');
+                    if let Some((name, _)) = name {
+                        if bucket == Bucket::Symbolic {
+                            if let Some(name) = name.strip_suffix("-symbolic") {
+                                icons
+                                    .entry(name.to_owned())
+                                    .or_default()
+                                    .insert(bucket, file.path());
+                            }
+                        } else {
+                            icons.entry(name.to_owned()).or_default().insert(bucket, file.path());
+                        }
+                    }
+                }
+            }
+        }
+
+        // This path is hardcoded in the specification.
+        if let Ok(read_dir) = fs::read_dir("/usr/share/pixmaps") {
+            for file in read_dir {
+                let file = file.unwrap();
+
+                let file_name = file.file_name();
+                let file_name = match file_name.to_str() {
+                    Some(file_name) => file_name,
+                    None => continue,
+                };
+
+                match file_name.rsplit_once('.') {
+                    Some((name, "svg")) => {
+                        icons
+                            .entry(name.to_owned())
+                            .or_default()
+                            .insert(Bucket::Scalable, file.path());
+                    },
+                    Some((name, "png")) => {
+                        // We don’t have any information about the size of the icon here.
+                        icons
+                            .entry(name.to_owned())
+                            .or_default()
+                            .insert(Bucket::Unknown, file.path());
+                    },
```

Comment:
I don't think `Unknown` is a good name here. Just call it bitmap or something. We know the image type we just don't know its size.